controllers/contacts.js

```diff
@@ -0,0 +1,86 @@
+const { Router } = require('express');
+const { Contact } = require('../models');
+const { dropUndefined } = require('../utils/dropUndefined');
+const { buildFilter } = require('../utils/queryHelper');
+const ObjectId = require('mongodb').ObjectID;
+
+const router = Router();
+
+// gets one contact by contact ID
+router.get('/:id', async (req, res) => {
+	const { id } = req.params;
+	let contactId = ObjectId(id);
+
+	try {
+		const contact = await Contact.findById(contactId);
+
+		if (!contact) {
+			res.status(404).send({ error: 'Contact not found!' });
+		} else {
+			res.json(contact);
+		}
+	} catch (err) {
+		res.status(500).send({ error: "Something went wrong; we're on it!" });
+	}
+});
+
+// gets all contacts associated with a user
+router.get('/userid/:id', async (req, res) => {
+	const { id } = req.params;
+	let contactId = ObjectId(id);
+
+	try {
+		const contact = await Contact.find({ userId: contactId });
```

Comment:
The conversion to `ObjectId` for the inputed `userId` could be unnecessary here.

Reply:
I was having issues with that last night. Mongoose was returning null when I searched with a string ID and it seemed to only be working after the conversion to ObjectID. However, I'll go through it again in a second. Its definitely possible that something else was amiss

Reply:
nevermind im a kook. Fixing it right now

Reply:
Weird, any idea what could've been wrong last night?

Reply:
I'm not entirely sure but it had to be some unrelated issue that I caught later on. Good catch on that